bin/tx-prover/Cargo.toml

```diff
@@ -11,30 +11,49 @@ repository.workspace = true
 rust-version.workspace = true
 edition.workspace = true
 
+[[bin]]
+name = "prover-service"
+path = "src/main.rs"
+required-features = ["std"]
+
+[lib]
+crate-type = ["lib"]
+
 [features]
-testing = ["miden-objects/testing", "miden-lib/testing", "miden-tx/testing"]
+async = ["miden-tx/async"]
+default = ["std"]
+std = ["miden-objects/std", "miden-tx/std", "dep:tokio", "dep:tonic-web", "dep:tokio-stream", "dep:axum",  "dep:tracing", "dep:tracing-subscriber", "tonic/transport"]
+
+[target.'cfg(all(target_arch = "wasm32", target_os = "unknown"))'.dependencies]
+tonic-web-wasm-client = { version = "0.6", default-features = false }
+tonic = { version = "0.12", default-features = false, features = ["prost", "codegen"] }
+getrandom = { version = "0.2", features = ["js"], optional = true }
+
+[target.'cfg(not(all(target_arch = "wasm32", target_os = "unknown")))'.dependencies]
+tonic = { version = "0.12", default-features = false, features = ["prost", "codegen", "transport"] }
```

Comment:
Question: does `target_os = "unknown"` basically mean any OS?

Reply:
If we wanted to mean any OS, we would not specify `target_os` (ie, `[target.'cfg(all(target_arch = "wasm32"))'.dependencies]`). I could be wrong, but I think `target_os = "unknown"` basically means any environment "without" OS. That is, for instance, for WASI environments this dependencies separator would not match. I added the `target_os` because of the `tonic-web-wasm-client` doc which mentions it supports this target explicity, but I think it would not matter much in practice if we removed it.